connectors/file/file-hdfs/src/main/java/org/teiid/hdfs/HdfsConnection.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags and
+ * the COPYRIGHT.txt file distributed with this work.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.teiid.hdfs;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Vector;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.io.IOUtils;
+import org.teiid.core.TeiidRuntimeException;
+import org.teiid.file.VirtualFile;
+import org.teiid.file.VirtualFileConnection;
+import org.teiid.translator.TranslatorException;
+
+public class HdfsConnection implements VirtualFileConnection {
+
+    //the filesystem is not thread-safe, so one is needed per connection
+    private final FileSystem fileSystem;
+
+    public HdfsConnection(HdfsConfiguration config) throws TranslatorException {
+        this.fileSystem = createFileSystem(config.getFsUri(), config.getResourcePath());
+    }
+
+    protected FileSystem createFileSystem(String fsUri, String resourcePath) throws TranslatorException {
+        Configuration configuration = new Configuration();
+        if(resourcePath != null){
+            if (HdfsConnection.class.getResourceAsStream(resourcePath) != null) {
```

Comment:
What is this extra check for?

Reply:
For wildfly we don't want to assume that any additional configuration resource will be on the classpath.  It's much easier for it to be at a file system path.  The existing addResource method only looks in the classpath.